core/templates/services/insert-script.service.ts

```diff
@@ -22,41 +22,86 @@ import {downgradeInjectable} from '@angular/upgrade/static';
 export enum KNOWN_SCRIPTS {
   DONORBOX = 'DONORBOX',
   UNKNOWN = 'UNKNOWN',
+  MATHJAX = 'MATHJAX',
 }
 
 @Injectable({
   providedIn: 'root',
 })
 export class InsertScriptService {
-  private loadedScripts: Set<string> = new Set<string>();
+  // Set of scripts that have already loaded.
+  private fullyLoadedScripts: Set<string> = new Set<string>();
+  // Maps scripts that are currently still loading along with their on-load-callback promises.
+  private partiallyLoadedScripts: Map<string, Promise<void>> = new Map();
   private renderer: Renderer2;
 
   constructor(rendererFactory: RendererFactory2) {
     this.renderer = rendererFactory.createRenderer(null, null);
   }
 
-  loadScript(script: KNOWN_SCRIPTS): boolean {
-    if (this.loadedScripts.has(script)) {
+  isScriptLoaded(script: KNOWN_SCRIPTS): boolean {
+    if (this.fullyLoadedScripts.has(script)) {
```

Comment:
Perhaps just return this as it is a boolean value.

Reply:
Done